src/gausskernel/runtime/executor/nodeModifyTable.cpp

```diff
@@ -992,7 +992,12 @@ TupleTableSlot* ExecInsertT(ModifyTableState* state, TupleTableSlot* slot, Tuple
                                 searchHBucketFakeRelation(estate->esfRelations, estate->es_query_cxt,
                                     result_relation_desc, bucket_id, target_rel);
                             }
-                            new_id = tableam_tuple_insert(target_rel, tuple, estate->es_output_cid, 0, NULL);
+                            
+                            if (IsK2PgRelation(target_rel)) {
```

Comment:
Could do a more generic update at heap_insert method in heapam.cpp 

https://github.com/futurewei-cloud/chogori-opengauss/commit/6db1db72b47a294b48cdedf6d37fd88110f51fd0#diff-07e6dc90be18e26d222004fc6c439178c7c68cdd8e4e43690ca04af501e35021

Reply:
I forgot to check the link so just moved the code. It looks same except the warning. Not sure if we should keep the warning.
```
PostgreSQL stand-alone backend 9.2.4
backend> INSERT INTO pg_depend SELECT 0,0,0, tableoid,oid,0, 'p' FROM pg_ts_config;
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.292 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.354 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.362 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.370 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
backend>
```

Reply:
let us remove the warning or use elog as debug level log